google/auth/credentials_async.py

```diff
@@ -62,7 +63,10 @@ async def before_request(self, request, method, url, headers):
         # the http request.)
 
         if not self.valid:
-            self.refresh(request)
+            if inspect.iscoroutinefunction(self.refresh):
```

Comment:
what is the scenario where this wouldn't be a coroutine?

Reply:
There are a couple of async credentials scenarios where self.refresh for those class instances return a None Type object (when it is not an async function). We run into this in an auth/jwt credentials object, and therefore add the check on whether refresh for that specific class instance is a coroutine or not.